Task 18: **"Explain what the test result allows you to claim and what it does not allow you to claim."** (from the Day 2 daily task list in your plan, paired with task 17, which was running the t-test itself).

In [15]:
pip install kagglehub

Defaulting to user installation because normal site-packages is not writeable
Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 26.2 -> 26.2.1
[notice] To update, run: python.exe -m pip install --upgrade pip


In [16]:
import kagglehub

# Download latest version
path = kagglehub.dataset_download("blastchar/telco-customer-churn")

print("Path to dataset files:", path)

Path to dataset files: C:\Users\DESK0059-\.cache\kagglehub\datasets\blastchar\telco-customer-churn\versions\1


In [17]:
import pandas as pd
import os 


csv_file = os.path.join(path,'WA_Fn-UseC_-Telco-Customer-Churn.csv')
df = pd.read_csv(csv_file)
df['Churn'].value_counts()

Churn
No     5174
Yes    1869
Name: count, dtype: int64

In [ ]:
pip install scipy

In [20]:
from scipy import stats

churned_tenure = df[df["Churn"] == "Yes"]["tenure"]
non_churned_tenure = df[df["Churn"] == "No"]["tenure"]

t_stat, p_value = stats.ttest_ind(non_churned_tenure, churned_tenure)

print(f"t-statistic: {t_stat:.3f}")
print(f"p-value: {p_value:.5f}")
print(f"Non-churned avg tenure: {non_churned_tenure.mean():.2f}")
print(f"Churned avg tenure: {churned_tenure.mean():.2f}")

t-statistic: 31.580
p-value: 0.00000
Non-churned avg tenure: 37.57
Churned avg tenure: 17.98


### Task 18 — Interpretation of the T-Test Result

**What we can claim:**
- There is a statistically significant difference in average tenure between churned and non-churned customers (t = 31.58, p < 0.00001)
- Non-churned customers have meaningfully longer tenure, this isn't random noise, the large sample size (7043) and near-zero p-value make that clear
- Tenure looks like a strong candidate feature for predicting churn

**What we cannot claim:**
- That low tenure *causes* churn, this is correlation, not causation (both could be driven by something else, e.g. contract type — newer customers may disproportionately be on month-to-month plans, which independently drives churn)
- That every churned customer has low tenure, this is a difference in *averages*, individual customers vary a lot
- That tenure alone can predict churn well, a t-test only compares two groups' means; it says nothing about how good a full predictive model would be